tests/src/gui/testqgsvaluerelationwidgetwrapper.cpp

```diff
@@ -624,9 +646,181 @@ void TestQgsValueRelationWidgetWrapper::testWithJsonInGPKG()
 #endif
 }
 
-void TestQgsValueRelationWidgetWrapper::testWithJsonInSpatialite()
+// Same test procedure like in testWithJsonInGPKG to check the non-json way of storing multi-selections into formatted strings
+void TestQgsValueRelationWidgetWrapper::testWithTextInGPKG()
 {
+  // create ogr gpkg layers
+  QString myFileName( TEST_DATA_DIR ); //defined in CmakeLists.txt
+  QString myTempDirName = tempDir.path();
+  QFile::copy( myFileName + "/provider/test_json.gpkg", myTempDirName + "/test_json.gpkg" );
```

Comment:
I think that for extra safety you may add test cases with `{},"'` in author names to this test file.

Also, if isn't there a PG test yet, we should prepare a table with string PK and  `{},"'` in the PK values and check it with both JSON and hstore field types.